lld/wasm/SymbolTable.cpp

```diff
@@ -310,9 +310,14 @@ static bool shouldReplace(const Symbol *existing, InputFile *newFile,
   }
 
   // Neither symbol is week. They conflict.
```

Comment:
It looks like `reportDuplicate` in the ELF linker simple returns early when `allowMultipleDefinition` is set.  I guess we should do the same?

Reply:
If I'm not mistaken, it still reports the warning somewhere, therefore I left the warning message here since it's the only place where we check for double definitions.

Reply:
I see that ELF has some checks with `errorOrWarn`; duplicated symbols are one of them. Would you like me to do the same for `wasm-ld`? In that case, I may need some help with other checks, which may be non-breaking in case of a warning.

Reply:
To follow the ELF logic, I had to add the `--noinhibit-exec` flag that is checked in `errorOrWarn`. Currently `errorOrWarn` is only used for multiple definitions. It can be used elsewhere if needed.